src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1256,11 +1334,12 @@ def _retrieve_logit_processors(
 
         if is_shortform and generation_config.forced_decoder_ids is not None:
             forced_tokens_proc = ForceTokensLogitsProcessor(generation_config.forced_decoder_ids)
```

Comment:
We're leaving the `ForceTokensLogitsProcessor` here for backwards compatibility for short-form. But it should be removed in 4.39.